### DISCUSSION TITLE
Generate a bare SPA?

### DISCUSSION THREAD

**Discussion by yooneskh** [nuxt/nuxt#18104] [Questions]
Hello everyone and thanks for your fantastic work.

I am looking for ways to unify my codebases for SPA, SSR, Hybrid mobile (through capacitor) and desktop apps (through tauri). All of these are possible with a simple vue app instead of the SSR obviously 

My question is that will nuxt 3 have a way to generate a completely bare SPA version of the app? like a normal vite (or vue-cli) build? if so, how will the server plugins and `useAsyncData` and such behave? are they (or will they be) documented?

**Comment by lucpotage** [comment]
There is an ongoing RFC about client-side rendering: https://github.com/nuxt/framework/discussions/560
Everything you need to know about `asyncData` is in this article: https://nuxtjs.org/docs/features/data-fetching/#async-data
What's new in Nuxt 3 and `useAsyncData` is here: https://v3.nuxtjs.org/docs/usage/data-fetching#useasyncdata

**Reply by yooneskh** [reply]
Thanks for your response @lucpotage . So in general, there will be a mode of build for Nuxt that will generate a simple SPA, right?

**Reply by lucpotage** [reply]
I suppose based on what I’ve read and considering it was possible in Nuxt 2, but only maintainers can confirm.

**Reply by yooneskh** [reply]
yeah i was pretty sure it would be possible too. just wanted to get a confirmation. thank you
